linera-client/src/chain_listener.rs

```diff
@@ -167,12 +167,15 @@ impl ChainListener {
         let admin_listener: NotificationStream = if client.admin_id() == chain_id {
             Box::pin(stream::pending())
         } else {
-            Box::pin(client.subscribe_to(client.admin_id()).await?)
+            let stream = client.subscribe_to(client.admin_id()).await?;
+            Box::pin(stream.filter(|notification: &Notification| {
```

Comment:
Does the future need to be pinned?

Reply:
Yes, the `fuse()` call below seems to require that.